x-pack/plugins/maps/public/components/layer_addpanel/index.js

```diff
@@ -7,43 +7,46 @@
 import { connect } from 'react-redux';
 import { AddLayerPanel } from './view';
 import { getFlyoutDisplay, updateFlyout, FLYOUT_STATE } from '../../store/ui';
-import { getTemporaryLayers, getMapColors } from '../../selectors/map_selectors';
+import { getSelectedLayer, getMapColors } from '../../selectors/map_selectors';
 import { getInspectorAdapters } from '../../store/non_serializable_instances';
 import {
-  addLayer,
   removeLayer,
-  clearTemporaryLayers,
+  setTransientLayer,
+  addLayer,
   setSelectedLayer,
+  removeTransientLayer
 } from '../../actions/store_actions';
-import _ from 'lodash';
 
 function mapStateToProps(state = {}) {
-
-  function isLoading() {
-    const tmp = getTemporaryLayers(state);
-    return tmp.some((layer) => layer.isLayerLoading());
-  }
+  const selectedLayer = getSelectedLayer(state);
   return {
     inspectorAdapters: getInspectorAdapters(state),
     flyoutVisible: getFlyoutDisplay(state) !== FLYOUT_STATE.NONE,
-    layerLoading: isLoading(),
-    temporaryLayers: !_.isEmpty(getTemporaryLayers(state)),
+    hasLayerSelected: !!selectedLayer,
+    isLoading: selectedLayer && selectedLayer.isLayerLoading(),
     mapColors: getMapColors(state),
   };
 }
 
 function mapDispatchToProps(dispatch) {
   return {
-    closeFlyout: () => {
+    closeFlyout: layerId => {
       dispatch(updateFlyout(FLYOUT_STATE.NONE));
-      dispatch(clearTemporaryLayers());
+      dispatch(removeTransientLayer());
+      dispatch(setSelectedLayer(null));
+      dispatch(removeLayer(layerId));
```

Comment:
This is not needed. `removeTransientLayer` already dispatches `removeLayer `.

`closeFlyout` is a duplicate of `clearTransientLayerStateAndCloseFlyout ` and should just dispatch that instead

Reply:
In fact, `clearTransientLayerStateAndCloseFlyout` should just renamed to `closeFlyout` and moved to store/ui

Reply:
I removed the extra `removeLayer`, that was a good call. `closeFlyout` and `clearTransientLayerStateAndCloseFlyout` evolved separately to the same point. Nice catch! I am leaving it where it is for now though as it's also called on save in the controller. Of course I could make two calls from each of these locations, one to clear transient layers, the next to close the flyout but I think this works reasonably well.